File: core/src/main/java/forpleuvoir/dhwu1a/core/message/base/MessageType.java
package forpleuvoir.dhwu1a.core.message.base;

import forpleuvoir.dhwu1a.core.message.messagesender.*;

/**
 * @author forpleuvoir
 * <p>#project_name dhwu1a
 * <p>#package forpleuvoir.dhwu1a.core.message.base
 * <p>#class_name MessageType
 * <p>#create_time 2021/6/29 22:35
 */
public enum MessageType {
    GroupMessage(GroupMessageSender.class),
    FriendMessage(FriendMessageSender.class),
    TempMessage(TempMessageSender.class),
    StrangerMessage(StrangerMessageSender.class),
    OtherClientMessage(OtherClientMessageSender.class);

    private final Class<? extends MessageSender> clazz;

    MessageType(Class<? extends MessageSender> clazz) {
        this.clazz = clazz;
    }

    public Class<? extends MessageSender> getClazz() {
        return clazz;
    }
}

File: core/src/main/kotlin/forpleuvoir/dhwu1a/core/message/base/MessageType.kt
package forpleuvoir.dhwu1a.core.message.base

import forpleuvoir.dhwu1a.core.message.messagesender.*

/**
 * @author forpleuvoir
 *
 * #project_name dhwu1a
 *
 * #package forpleuvoir.dhwu1a.core.message.base
 *
 * #class_name MessageType
 *
 * #create_time 2021/6/29 22:35
 */
enum class MessageType(val clazz: Class<out MessageSender?>) {
    GroupMessage(GroupMessageSender::class.java),
    FriendMessage(FriendMessageSender::class.java),
    TempMessage(TempMessageSender::class.java),
    StrangerMessage(StrangerMessageSender::class.java),
    OtherClientMessage(OtherClientMessageSender::class.java);
}